src/module/item/entity.js

```diff
@@ -68,49 +68,48 @@ export class OseItem extends Item {
     return itemData;
   }
 
-  rollWeapon(options = {}) {
-    let isNPC = this.actor.type != "character";
-    const targets = 5;
+  async getAttackType(options) {
+    if (this.actor.type != "character") return CONFIG.OSE.attack_types.attack;
     const itemData = this.system;
-
-    let type = isNPC ? "attack" : "melee";
-    const rollData = {
-      item: this._source,
-      actor: this.actor,
-      roll: {
-        save: itemData.save,
-        target: null,
-      },
-    };
-
-    if (itemData.missile && itemData.melee && !isNPC) {
-      // Dialog
-      new Dialog({
-        title: "Choose Attack Range",
-        content: "",
-        buttons: {
-          melee: {
-            icon: '<i class="fas fa-fist-raised"></i>',
-            label: "Melee",
-            callback: () => {
-              this.actor.targetAttack(rollData, "melee", options);
-            },
+    if (itemData.missile && itemData.melee) {
+      if (CONFIG.OSE.attack_types[options.preferedType]){
+        return CONFIG.OSE.attack_types[options.preferedType];
```

Comment:
preferredType, extra r

Spelling is hard! I wouldn't mention it if there wasn't something else to change